### DISCUSSION TITLE
Aprendendo JS em 2019

### DISCUSSION THREAD

**Discussion by Rhamon12** [frontendbr/forum#1421] [Discussão]
Olá, pessoal estou com uma dúvida sobre como iniciar meus estudos em desenvolvimento web.

Primeiramente gostaria de saber quais livros, cursos gratuitos e ferramentas vocês que tem experiência recomendam.

Em segundo lugar estou quase zerado de conhecimento, curso ADS mas em si trabalho e estudo acabo não tendo muito tempo para estudar em casa mas o tempo que tenho estudo, ultimamente estudo Java  e quero ser um desenvolvedor full-stack, e preciso dividir meu tempo de estudo para desenvolvimento web onde também acho interessante mas tenho dúvidas se atualmente mudou a forma que se estuda tal linguagem como javascript.

Adendo, gostaria de começar do zero e me aprofundar o máximo possível não importa o tempo que demore, sou muito grato a qualquer ajuda vinda de vocês, também procurei tópicos parecidos com esse mas não me esclareceram muito sobre pois também tenho dúvidas sobre o nosso ano atual.

**Comment by MrS4w** [comment]
Até final do ano passado não manjava nada de JS, [esse curso](https://www.udemy.com/course/curso-web/) me deu uma boa base, atualmente trabalho com TS que é praticamente um ES6 tipado :smile: .
Eu poderia listar milhares de coisas boas por aí, mas esse daí eu recomendo pra quem tá começando do zero.

**Comment by luizfelipehh** [comment]
@Rhamon12 O conselho que dou é, procure cursos em plataformas online(udemy,alura,udacity).

Mas antes, deve procurar e entender qual Framework irá te ajudar para front-end e qual irá te ajudar para back-end.

Só praticando irá pegar experiência, não tem outro caminho.

**Comment by Rhamon12** [comment]
> Até final do ano passado não manjava nada de JS, [esse curso](https://www.udemy.com/course/curso-web/) me deu uma boa base, atualmente trabalho com TS que é praticamente um ES6 tipado 😄 .
> Eu poderia listar milhares de coisas boas por aí, mas esse daí eu recomendo pra quem tá começando do zero.

Realmente interessante, muito obrigado pela recomendação.


> @Rhamon12 O conselho que dou é, procure cursos em plataformas online(udemy,alura,udacity).
> 
> Mas antes, deve procurar e entender qual Framework irá te ajudar para front-end e qual irá te ajudar para back-end.
> 
> Só praticando irá pegar experiência, não tem outro caminho.

Muito obrigado pelos conselhos, irei buscar esses Framework´s agora.

**Comment by MrS4w** [comment]
@Rhamon12 Não esqueça do JS puro antes de qualquer framework/lib :nerd_face:

**Comment by praganavictor** [comment]
Uma coisa que eu acho que vc pode precisar é exercitar os metodos que vai aprender do JS e tem esse curso aqui https://javascript30.com em ingles que vai te colocar para fazer 30 projetinhos em JS que são muito legais para praticar (se o ingles nao for uma barreira é claro)

**Comment by vicainelli** [comment]
Comecei a ver uns vídeos da [Rocketseat no YouTube](https://www.youtube.com/channel/UCSfwM5u0Kce6Cce8_S72olg/playlists) e achei bem boa a didática, olhei alguns poucos vídeos no [site](https://rocketseat.com.br/starter/curso-gratuito-javascript) lá tem uma espécie de trilha pra você seguir e tal.

**Comment by onlurking** [comment]
Recomendo pegar a fundação primeiro, o melhor tutorial de HTML e CSS que conheço de longe é o do [Interneting Is Hard](https://internetingishard.com/html-and-css), e junto com ele usar os sites [htmlreference](http://htmlreference.io) e [cssreference](https://cssreference.io).

Sobre CSS: aprenda como funciona especificidade, escreva apenas classes(nunca id, e evitar ao máximo tag selector), use [flexbox](https://origamid.com/projetos/flexbox-guia-completo) pra layout e se possível BEM(Block Element Modifier, que é apenas uma forma de nomear classes em CSS), vai lhe poupar MUITA dor de cabeça quando estiver mexendo com CSS.

Sobre JS: Um guia de estudos bem maneiro pra JS Puro é do [Learn Vanilla JS](https://learnvanillajs.com/roadmap), principalmente a parte dos projetos, eles também enviam [umas dicas](https://gomakethings.com/articles) por e-mail e há um guia de métodos pra [referência](https://vanillajstoolkit.com/reference). Outro é o curso do Wes Bos que recomendaram acima(Javascript30), tem uma pancada de exercícios.

A [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide) é absurdamente boa, sempre use como referência quando der. Outro lugar que tem umas coisas legais é o site do [Flavio Copes](https://flaviocopes.com/javascript).

**Comment by gabriel-hahn** [comment]
No início da aprendizagem acho legal tu pegar aqueles cursos mais "gerais" que vão te dar uma ampla visão de como tudo funciona. Curto bastante os cursos da Udemy e o curso citado pelo @MrS4w é uma pedida, a didática é bem boa.
Depois desse geral, existem milhares de cursos na Udemy também e em algumas outras plataformas como Alura e Youtube que vão te orientar e te dar uma profundidade nos assuntos, envolvendo frameworks e JavaScript mais avançado. Para a parte do JS, recomendo MUITO [este curso](https://www.udemy.com/understand-javascript/) (se o inglês não for um problema).
Procura focar mais em HTML, CSS e JavaScript agora no início. Não apressa o aprendizado, tenta entender o básico de como tudo funciona e entender os conceitos, depois disso tu parte para o aprendizado de Frameworks que vão "facilitar" tua vida no desenvolvimento de aplicações mais complexas. Boa sorte! :)

**Comment by allangrds** [comment]
### Livros
- [Eloquent Javascript](https://github.com/braziljs/eloquente-javascript)
- [You Don't Know JS](https://github.com/cezaraugusto/You-Dont-Know-JS)

### Vídeos
- [Desvendando a linguagem JavaScript](https://www.youtube.com/watch?v=093dIOCNeIc&list=PLQCmSnNFVYnT1-oeDOSBnt164802rkegc)

### Cursos
- [JavaScript: Understanding the Weird Parts](https://www.udemy.com/understand-javascript/): esse não é um curso com conceitos recentes, mas aborda vários tópicos como a parte léxica da linguagem, e o básico dela. As coisas mais novas você consegue ir pegando em blog, mas esse curso te dá um entendimento muito legal.

**Comment by betocostadev** [comment]
> Comecei a ver uns vídeos da [Rocketseat no YouTube](https://www.youtube.com/channel/UCSfwM5u0Kce6Cce8_S72olg/playlists) e achei bem boa a didática, olhei alguns poucos vídeos no [site](https://rocketseat.com.br/starter/curso-gratuito-javascript) lá tem uma espécie de trilha pra você seguir e tal.

Sabe me dizer sobre o Bootcamp deles, se é bom e o valor?
Andei pesquisando, mas não encontrei. Me cadastrei na plataforma pra testar um pouco. Achei legal, mas não vi o suficiente pra ter certeza sobre eles, ao menos para os cursos gratuitos, parece que é bem 'raso' o que eles passam.

**Comment by vicainelli** [comment]
@betocostadev então, eu não fiz também, mas falei com um cara num meetup que fez e ele curtiu.
Se não me engano o valor era entre R$ 1200,00 e R$ 1500,00.

**Comment by betocostadev** [comment]
@vicainelli, obrigado!
Bom, se o curso for bom mesmo, é de se pensar.
Vou terminar alguns cursos que estou fazendo e depois darei uma olhada. Ultimamente, pelo que ando pesquisando em SP, sem Node e React está ficando complicado o mercado.
